src/bsl/transaction.rs

```diff
@@ -41,8 +41,12 @@ impl<'a> Visit<'a> for Transaction<'a> {
                     slice: &slice[..consumed],
                     inputs_outputs_len: NonZeroU32::new(inputs_outputs_len as u32), // inputs_outputs_len is at least 2 bytes if both empty, they contain the compact int len
                 };
-                visit.visit_transaction(&tx);
-                Ok(ParseResult::new(&slice[consumed..], tx))
+                let stop = visit.visit_transaction(&tx);
+                if stop {
+                    Err(Error::VisitBreak)
```

Comment:
should breaking return an error? it makes sense to branch (`?` operator of `Result`) on a parsing error, but breaking is intentionally done by the user

Reply:
yeah it's not very conventional, however, I don't want to change the Result type for this edge case because it would impact all the visitor implementors